app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js

```diff
@@ -65,9 +65,16 @@ async function switchEthereumChainHandler(
 
   const { origin } = req;
 
-  const { chainId } = req.params[0];
+  const { chainId, networkConfigurationId } = req.params[0];
```

Comment:
The second param I've added here is for internal use (not part of the public API). It is used by the `addEthereumChain` handler [here](https://github.com/MetaMask/metamask-extension/pull/17421/files#diff-b3485a1ea9943fd380364f73a53295289f2ccd3817423d4f245b5102f8620b3cR170) and [here](https://github.com/MetaMask/metamask-extension/pull/17421/files#diff-b3485a1ea9943fd380364f73a53295289f2ccd3817423d4f245b5102f8620b3cR314) to pass a single value to the subsequent `switchEthereumChain` call that can be used [to quickly switch network](https://github.com/MetaMask/metamask-extension/pull/17421/files#diff-93f17dab5026baf840aec766ac5b6ebcd35d5bb0abb45e2c3f41791423f2a488R71) without having [to again loop through all the networkConfigurations](https://github.com/MetaMask/metamask-extension/pull/17421/files#diff-93f17dab5026baf840aec766ac5b6ebcd35d5bb0abb45e2c3f41791423f2a488R29) and determine which network we want to prompt the user to switch to.

Reply:
Hmm, I'm having trouble following this. I didn't think that the `addEthereumChain` invoked this code at all - it seems to switch chains directly (see the `setActiveNetwork` calls)

Reply:
Yep. I was confused about how/whent he handler was invoked:  I've removed this change here: https://github.com/MetaMask/metamask-extension/pull/17421/commits/73ade0f9f23586ce185a52a32991fadef8f2cea0

Reply:
Yep! I misunderstood how/when the these handlers were invoked. Fixed/removed here: https://github.com/MetaMask/metamask-extension/pull/17421/commits/ea4bf3dbbc31bab3a25f51e4902338b64bce9ccc